lib/methods/ifwapichecktask.cpp

```diff
@@ -0,0 +1,314 @@
+			checkable, command, cr,
+			"Got JSON, but not an object, from IfW API on host '" + psHost + "' port '" + psPort + "'", start, end
+		);
+		return;
+	}
+
+	Value jsonBranch;
+
+	if (!Dictionary::Ptr(jsonRoot)->Get(psCommand, &jsonBranch)) {
+		ReportIfwCheckResult(
+			checkable, command, cr,
+			"Missing ." + psCommand + " in JSON object from IfW API on host '" + psHost + "' port '" + psPort + "'", start, end
+		);
+		return;
+	}
+
+	if (!jsonBranch.IsObjectType<Dictionary>()) {
+		ReportIfwCheckResult(
+			checkable, command, cr,
+			"." + psCommand + " is not an object in JSON from IfW API on host '" + psHost + "' port '" + psPort + "'", start, end
+		);
+		return;
+	}
+
+	Dictionary::Ptr result = jsonBranch;
+	double exitcode;
+
+	try {
+		exitcode = result->Get("exitcode");
+	} catch (const std::exception& ex) {
+		ReportIfwCheckResult(
+			checkable, command, cr,
+			"Got bad exitcode from IfW API on host '" + psHost + "' port '" + psPort + "': " + ex.what(), start, end
+		);
+		return;
+	}
+
+	Array::Ptr perfdata;
+
+	try {
+		perfdata = result->Get("perfdata");
+	} catch (const std::exception& ex) {
+		ReportIfwCheckResult(
+			checkable, command, cr,
+			"Got bad perfdata from IfW API on host '" + psHost + "' port '" + psPort + "': " + ex.what(), start, end
+		);
+		return;
+	}
+
+	ReportIfwCheckResult(checkable, command, cr, result->Get("checkresult"), start, end, exitcode, perfdata);
+}
+
+static void DoIfwNetIo(
+	boost::asio::yield_context yc, const Checkable::Ptr& checkable, const CheckCommand::Ptr& command,
+	const CheckResult::Ptr& cr, const String& psCommand, const String& psHost, const String& psPort,
+	boost::beast::http::request<boost::beast::http::string_body>& req
+)
+{
+	using namespace boost::asio;
+	using namespace boost::beast;
+	using namespace boost::beast::http;
+
+	ssl::context ctx (ssl::context::tls);
+	AsioTlsStream conn (IoEngine::Get().GetIoContext(), ctx);
+	flat_buffer buf;
+	auto resp (Shared<response<string_body>>::Make());
+	double start = Utility::GetTime();
+
+	try {
+		Connect(conn.lowest_layer(), psHost, psPort, yc);
+	} catch (const std::exception& ex) {
+		ReportIfwCheckResult(
+			yc, checkable, command, cr,
+			"Can't connect to IfW API on host '" + psHost + "' port '" + psPort + "': " + ex.what(), start
+		);
+		return;
+	}
+
+	try {
+		conn.next_layer().async_handshake(conn.next_layer().client, yc);
+	} catch (const std::exception& ex) {
+		ReportIfwCheckResult(
+			yc, checkable, command, cr,
+			"TLS handshake with IfW API on host '" + psHost + "' port '" + psPort + "' failed: " + ex.what(), start
+		);
+		return;
+	}
+
+	try {
+		async_write(conn, req, yc);
+		conn.async_flush(yc);
+	} catch (const std::exception& ex) {
+		ReportIfwCheckResult(
+			yc, checkable, command, cr,
+			"Can't send HTTP request to IfW API on host '" + psHost + "' port '" + psPort + "': " + ex.what(), start
+		);
+		return;
+	}
+
+	try {
+		async_read(conn, buf, *(response<string_body>*)resp.get(), yc);
```

Comment:
Why does this require a poiner cast? Anyways, same with `std::move` in the lambda capture applies to `resp` in this function as well, that might just entirely remove the pointer.